STEP/AST/StandardEnvironment.cs

```diff
@@ -4,55 +4,242 @@ namespace STEP.AST;
+    {
+        Name = "Print",
+        ArduinoName = "Serial.println",
+        Parameters = new()
+        {
+            { "message", new Type { ActualType = TypeVal.String } }
+        },
+        Type = new Type { ActualType = TypeVal.Blank }
+    };
+    #endregion
+
+    #region Time
+    // delay(unsigned long)
+    public static StdFuncSymTableEntry Delay => new()
+    {
+        Name = "Wait",
+        ArduinoName = "delay",
+        Parameters = new()
+        {
+            { "ms", new Type { ActualType = TypeVal.Number } }
+        },
+        Type = new Type { ActualType = TypeVal.Blank }
+    };
+    #endregion
+
+    #region Math
+    public static StdFuncSymTableEntry Abs => new()
+    {
+        Name = "AbsoluteValue",
+        ArduinoName = "abs",
+        Parameters = new()
+        {
+            { "x", new Type { ActualType = TypeVal.Number } }
+        },
+        Type = new Type { ActualType = TypeVal.Number }
+    };
+
+    public static StdFuncSymTableEntry Constrain => new()
+    {
+        Name = "ConstrainValue",
+        ArduinoName = "constrain",
+        Parameters = new()
+        {
+            { "x", new Type { ActualType = TypeVal.Number } },
+            { "upperLimit", new Type { ActualType = TypeVal.Number } },
+            { "lowerLimit", new Type { ActualType = TypeVal.Number } }
+        },
+        Type = new Type { ActualType = TypeVal.Number }
+    };
+
+    public static StdFuncSymTableEntry Max => new()
+    {
+        Name = "Max",
+        ArduinoName = "max",
+        Parameters = new()
+        {
+            { "x", new Type { ActualType = TypeVal.Number } },
+            { "y", new Type { ActualType = TypeVal.Number } },
+        },
+        Type = new Type { ActualType = TypeVal.Number }
     };
 
-    // Analog I/O
-    // analagRead(), analogReference(), analogWrite()
+    public static StdFuncSymTableEntry Min => new()
+    {
+        Name = "Min",
+        ArduinoName = "min",
+        Parameters = new()
+        {
+            { "x", new Type { ActualType = TypeVal.Number } },
+            { "y", new Type { ActualType = TypeVal.Number } },
+        },
+        Type = new Type { ActualType = TypeVal.Number }
+    };
 
-    // I/O (Serial)
-    // print(), 
+    public static StdFuncSymTableEntry Power => new()
+    {
+        Name = "Power",
+        ArduinoName = "pow",
+        Parameters = new()
+        {
+            { "base", new Type { ActualType = TypeVal.Number } },
+            { "exponent", new Type { ActualType = TypeVal.Number } },
+        },
+        Type = new Type { ActualType = TypeVal.Number }
+    };
 
-    // Time
-    // delay(), delayMicroseconds(), micros(), millis()
+    public static StdFuncSymTableEntry Squared => new()
+    {
+        Name = "Squared",
+        ArduinoName = "sq",
+        Parameters = new()
+        {
+            { "x", new Type { ActualType = TypeVal.Number } },
+        },
+        Type = new Type { ActualType = TypeVal.Number }
+    };
```

Comment:
Er den ikke også lidt overflødig når vi har ^ operatoren?

Reply:
Tjoe, har bare taget alle de indbyggede Arduino-funktioner - det skader ikke at have begge dele 
![image](https://user-images.githubusercontent.com/71791903/165485808-a5a222c9-f45b-4248-81dc-2dd91b5dcfb7.png)

Reply:
i'll allow it